webservice/utils/get-data-from-form-section.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+
+var assign = require('es5-ext/object/assign');
+
+module.exports = function getDataFromSection(dataForm, result, typeMapper) {
+
+	if (!result) result = {};
+
+	var ownerType = dataForm.propertyMaster;
+
+	// first try toWsSchemaJson method
+	if (dataForm.toWebServiceSchemaJSON) {
+		var dataFormResult = dataForm.toWebServiceSchemaJSON(typeMapper);
+		return assign(result, dataFormResult);
+	}
+
+	// if tabular section
+	if (dataForm.propertyName) {
+
+		// exclude customs
+		if (!ownerType.get(dataForm.propertyName)
+				|| !ownerType.get(dataForm.propertyName).getItemType) {
+			return;
+		}
+
+		var formOwner = ownerType.get(dataForm.propertyName).getItemType();
+		if (formOwner.prototype.dataForms && formOwner.prototype.dataForms.size) {
+			formOwner.prototype.dataForms.map.forEach(function (subForm) {
+				result[dataForm.propertyName] = {};
+				result[dataForm.propertyName].type = "array";
+				result[dataForm.propertyName].items = { type: "object", properties: { } };
+
+				getDataFromSection(subForm, result[dataForm.propertyName].items.properties, typeMapper);
+
+			});
+		}
+	} else {
+		(dataForm.propertyNames || []).forEach(function (prop) {
```

Comment:
`dataForm.propertyNames` is redundant unless you plan for some `FormSection` not to have `toWebServiceSchemaJSON` - which would be rather bad idea. If we take this situation out of the picture you are left no way `dataForm.propertyNames` being defined. Anyway `dataForm` is rather not the best choice of name - it should rather be `section`.